extractor/filesystem/language/python/requirementsnet/requirements.go

```diff
@@ -40,34 +40,39 @@ const (
 
 // Config is the configuration for the Extractor.
 type Config struct {
-	filesystem.Extractor
+	// This should be an extractor to extract inventories from requirements.txt offline.
+	*requirements.Extractor // Extractor to extract inventories from requirements.txt offline.
 	resolve.Client
 }
 
 // DefaultConfig returns the default configuration for the extractor.
 func DefaultConfig() Config {
 	return Config{
-		Extractor: requirements.NewDefault(),
+		Extractor: requirements.NewDefault().(*requirements.Extractor),
 		Client:    resolution.NewPyPIRegistryClient(""),
 	}
 }
 
 // Extractor extracts python packages from requirements.txt files.
 type Extractor struct {
-	filesystem.Extractor
+	BaseExtractor *requirements.Extractor // The base extractor that we use to extract direct dependencies.
 	resolve.Client
 }
 
-// New returns a requirements.txt extractor.
+// New returns a requirements.txt transitive extractor.
 //
 // For most use cases, initialize with:
 // ```
 // e := New(DefaultConfig())
 // ```
 func New(cfg Config) *Extractor {
+	if cfg.Extractor.Name() != requirements.Name {
```

Comment:
Maybe add nil check here as well, for cases where they just pass in an empty config. I don't know whether the client would still work in those cases though.

Reply:
Actually, we don't even need this check, it's already enforced by the type. (nil check could still be a good idea though.